File: Sources/UsageWidget/Domain/PlanInfo.swift
import Foundation

/// Subscription plan info shown as a small badge on each provider card.
struct PlanInfo: Hashable, Codable, Sendable {
    let displayName: String   // "Max", "Pro", "Plus", "Team", "Enterprise"
    let rawValue: String?     // provider's raw plan_type, kept for debugging

    init(displayName: String, rawValue: String? = nil) {
        self.displayName = displayName
        self.rawValue = rawValue
    }

    /// Build a nicely-cased plan from a provider's raw `plan_type` string.
    static func from(rawPlanType raw: String?) -> PlanInfo? {
        guard let raw, !raw.trimmingCharacters(in: .whitespaces).isEmpty else { return nil }
        let pretty: String
        switch raw.lowercased() {
        case "free": pretty = "Free"
        case "plus": pretty = "Plus"
        case "pro": pretty = "Pro"
        case "team": pretty = "Team"
        case "enterprise", "ent": pretty = "Enterprise"
        case "max", "max_5x", "max_20x", "max5x", "max20x": pretty = "Max"
        case "individual_pro_plus", "pro_plus": pretty = "Pro+"
        case "individual_pro": pretty = "Pro"
        case "individual": pretty = "Individual"
        case "business": pretty = "Business"
        default: pretty = raw.replacingOccurrences(of: "_", with: " ").capitalized
        }
        return PlanInfo(displayName: pretty, rawValue: raw)
    }
}

File: windows/src/Core/Domain/PlanInfo.cs
using System.Globalization;

namespace LLMUsageWidget.Core.Domain;

/// <summary>Subscription plan info shown as a small badge on each provider card.</summary>
public sealed record PlanInfo(string DisplayName, string? RawValue = null)
{
    /// <summary>Build a nicely-cased plan from a provider's raw <c>plan_type</c> / org type string.</summary>
    public static PlanInfo? From(string? raw)
    {
        if (string.IsNullOrWhiteSpace(raw)) return null;
        string pretty = raw.ToLowerInvariant() switch
        {
            "free" => "Free",
            "plus" => "Plus",
            "pro" => "Pro",
            "team" => "Team",
            "enterprise" or "ent" => "Enterprise",
            "max" or "max_5x" or "max_20x" or "max5x" or "max20x" or "claude_max" => "Max",
            "claude_pro" => "Pro",
            "claude_free" => "Free",
            "claude_team" => "Team",
            "claude_enterprise" => "Enterprise",
            "individual_pro_plus" or "pro_plus" => "Pro+",
            "individual_pro" => "Pro",
            "individual" => "Individual",
            "business" => "Business",
            _ => CultureInfo.InvariantCulture.TextInfo.ToTitleCase(raw.Replace('_', ' ').ToLowerInvariant())
        };
        return new PlanInfo(pretty, raw);
    }
}
